### PR TITLE
refactor(icons): to use emotion styles

### PR DESCRIPTION
**Pull request by emmenko**
This is exiting!

The HOCs are now gone, maybe tree shaking works again now 🤓

### CODE REVIEW

File: src/components/icons/index.js [hunk 1]
```diff
@@ -106,323 +105,225 @@ import OrigVerifiedIcon from './svg/verified.react.svg';
...
-export { TagStackedIcon };
-export { TagIcon };
-export { TreeStructureIcon };
-export { TruckIcon };
-export { UserFilledIcon };
-export { UserLinearIcon };
-export { VerifiedIcon };
-export { WarningIcon };
-export { WorldIcon };
+export const AngleDownIcon = createStyledIcon(
```
Comment (emmenko) [inline src/components/icons/index.js]: We can export this directly now 🎉

File: src/components/icons/create-styled-icon.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      invariant(
+        theme,
+        `ui-kit/Icon: the specified theme '${theme}' is not supported.`
+      );
+      return 'inherit';
+    }
+  }
+};
+
+export default function createStyledIcon(Component, displayName) {
```
Comment (tdeekens) [inline src/components/icons/create-styled-icon.js]: Why do we need a factory function for this? We have quite a few of these in the app and they cause wasteful renders if I recall. Would it be an option to just have an `StyledIcon` component accepting an `iconComponent` prop and `iconDisplayName`?

Comment (emmenko) [reply src/components/icons/create-styled-icon.js]: The function is only used to create the component. The exported component is already styled, there are wasteful rerenders

**Comment by montezume** [comment]
How come the MoneyField tests are failing here?

**Comment by emmenko** [comment]
The scale size seems broken. I'll look into it

File: src/components/icons/create-styled-icon.js [hunk 3]
```diff
@@ -13,7 +13,7 @@ const getSizeStyle = size => {
   switch (size) {
     case 'scale':
       return `
-        svg.scale:not(:root) {
+        &:not(:root) {
```
Comment (montezume) [inline src/components/icons/create-styled-icon.js]: 👍

**Comment by montezume** [comment]
Excited to see if this helps with tree shaking 😇

**Pull request merged**